Helper/Api/Orders.php

```diff
@@ -1016,4 +1034,45 @@ private function calculateTaxClassesDiscounts(Item $item, Order $order): void
             }
         }
     }
+
+    /**
+     * @param Item $item
+     * @param Order $order
+     * @return void
+     */
+    private function logItemTaxAmount(Item $item, Order $order): void
+    {
+        $itemTaxAmount = $this->getPriceValueForPayload($item->getTaxAmount(), $order);
+        $this->logger->info('Order: #' . $order->getId() . ', item: SKU: "' . $item->getSku() . '", tax amount: ' . $itemTaxAmount);
+    }
+
+    /**
+     * @param Item $item
+     * @param Item $bundleItem
+     * @param Order $order
+     * @param bool $hasDiscount
+     * @param array|null $bundleItemWithDiscountData
+     * @return array
+     */
+    private function getPricePerUnitPayload(Item $item, Item $bundleItem, Order $order, bool $hasDiscount, ?array $bundleItemWithDiscountData): array
+    {
+        $bundleItemOriginalPrice = $bundleItem->getOriginalPrice();
+        $bundleItemSpecialPrice = $bundleItem->getPrice();
+        if (!$item->getTaxPercent() && $bundleItemOriginalPrice != $bundleItemSpecialPrice) {
+            $payload = [
+                'before_discount_ppu' => $this->getPriceValueForPayload($bundleItemOriginalPrice, $order),
+                'ppu' => $this->getPriceValueForPayload($bundleItemSpecialPrice, $order)
+            ];
+        } else {
+            $priceWithTax = $bundleItem->getPriceInclTax();
+            $bundleItemPriceWithTax = $priceWithTax ? $this->getPriceValueForPayload($bundleItem->getPriceInclTax(), $order) : 0.0;
+            $payload = [
+                'before_discount_ppu_wt' => $hasDiscount ? $bundleItemWithDiscountData['before_discount_ppu_wt'] : $bundleItemPriceWithTax,
+                'ppu_wt' => $hasDiscount ? $bundleItemWithDiscountData['ppu_wt'] : $bundleItemPriceWithTax
+            ];
+
+        }
```

Comment:
why on time you use the including price and another the excluding? which case it is?